src/components/Dropdown/V2/Dropdown.MenuContainer.tsx

```diff
@@ -217,9 +229,43 @@ export class MenuContainer extends React.PureComponent<Props> {
     items: Array<any>
     withIndex?: boolean
   }) => {
+    const {
+      allowMultipleSelection,
+      clearSelection,
+      focusItem,
+      selectionClearer,
+    } = this.props
+
+    if (allowMultipleSelection && selectionClearer) {
+      const clearerItem = {
+        value: selectionClearer,
+        id: createUniqueIDFactory('hsds-dropdown-v2-theallclearer')(),
```

Comment:
@tinkertrain Curious!

Can this ID factory be moved to the new `ItemSelectedCheck.tsx` component? If not, can the resulting function (from the factory) be cached at the top of this `MenuContainer`? And that fn be instantiated inline here

Reply:
Cached to the top, what I'm doing is creating a "special" item to prepend it to the rest of the items before we decide to use ItemSelectedCheck